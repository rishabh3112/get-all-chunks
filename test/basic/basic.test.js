const { getAllChunks } = require("../../lib/index.js");
const { resolve } = require("path");

const testPath = resolve(__dirname, "./src/code.js");
const folderPath = resolve(__dirname, "./src");

describe("basic", () => {
  it("should output correct chunks", async () => {
    const depTree = await getAllChunks(testPath, folderPath);
    expect(depTree).toMatchSnapshot();
  });
});
