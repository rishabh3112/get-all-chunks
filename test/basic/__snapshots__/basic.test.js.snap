// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic should output correct chunks 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "chunks": Set {
        "/Users/rishabh/Code/sprinklr/AST/test/basic/src/folderb/b.js",
        "/Users/rishabh/Code/sprinklr/AST/test/basic/src/folderc/c.js",
      },
      "path": "/Users/rishabh/Code/sprinklr/AST/test/basic/src/apple.js",
    },
  ],
  "chunks": Set {
    "/Users/rishabh/Code/sprinklr/AST/test/basic/src/folderc/c.js",
    "/Users/rishabh/Code/sprinklr/AST/test/basic/src/folderb/b.js",
  },
  "path": "/Users/rishabh/Code/sprinklr/AST/test/basic/src/code.js",
}
`;
